File: recursion/reversearray.cpp
#include<iostream>
using namespace std;

void revarr(int arr[],int i,int j){
    if(i>j) return ;
    int temp = arr[i];
     arr[i] = arr[j];
    arr[j] = temp;

    revarr(arr,i+1,j-1);
}


int main(){

    int arr[5] = {1,2,3,4,5};       
    int n=5;
    int i=0;
    int j = n-1;
    revarr(arr,i,j);
    for(int i =0;i<n;i++){
        cout<<arr[i];
    }
}

File: Injava/array/reversearray.java
package Injava.array;

public class reversearray {

    public static void main(String[] args) {
        int arr[]={1,2,3,4,5};
        int start=0;
        int end=arr.length-1;
        int temp=0;
        while(start<end){
            temp=arr[start];
            arr[start]=arr[end];
            arr[end]=temp;
            start++;
            end--;
        }
        for(int i=0;i<arr.length;i++){
            System.out.println(arr[i]);
        }
        
    }
    
}
